src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,447 @@
+  {
+    GEOSX_UNUSED_VAR( k );
+    real64 maxForce = 0;
+
+    CONSTITUTIVE_TYPE::KernelWrapper::DiscretizationOps::template fillLowerBTDB< numNodesPerElem >( stack.localJacobian );
+
+    constexpr int nPDof = 1;
+    constexpr int nUDof = numNodesPerElem * numDofPerTestSupportPoint;
+
+    for( int localNode = 0; localNode < numNodesPerElem; ++localNode )
+    {
+      for( int dim = 0; dim < numDofPerTestSupportPoint; ++dim )
+      {
+        localIndex const dof = LvArray::integerConversion< localIndex >( stack.localRowDofIndex[ numDofPerTestSupportPoint * localNode + dim ] - m_dofRankOffset );
+        if( dof < 0 || dof >= m_matrix.numRows() ) continue;
+        m_matrix.template addToRowBinarySearchUnsorted< parallelDeviceAtomic >( dof,
+                                                                                stack.localRowDofIndex,
+                                                                                stack.localJacobian[ numDofPerTestSupportPoint * localNode + dim ],
+                                                                                numNodesPerElem * numDofPerTrialSupportPoint );
+
+        RAJA::atomicAdd< parallelDeviceAtomic >( &m_rhs[ dof ], stack.localResidual[ numDofPerTestSupportPoint * localNode + dim ] );
+        maxForce = fmax( maxForce, fabs( stack.localResidual[ numDofPerTestSupportPoint * localNode + dim ] ) );
+
+        m_matrix.template addToRowBinarySearchUnsorted< parallelDeviceAtomic >( dof,
+                                                                                stack.localFlowDofIndex,
+                                                                                stack.localDispFlowJacobian[numDofPerTestSupportPoint * localNode + dim],
+                                                                                nPDof );
+
+      }
+    }
+
+    for( localIndex i = 0; i < nPDof; ++i )
+    {
+      localIndex const dof = LvArray::integerConversion< localIndex >( stack.localFlowDofIndex[ i ] - m_dofRankOffset );
+      if( dof < 0 || dof >= m_matrix.numRows() )
+        continue;
+      m_matrix.template addToRowBinarySearchUnsorted< parallelDeviceAtomic >( dof,
+                                                                              stack.localRowDofIndex,
+                                                                              stack.localFlowDispJacobian[i],
+                                                                              nUDof );
+      m_matrix.template addToRowBinarySearchUnsorted< parallelDeviceAtomic >( dof,
+                                                                              stack.localFlowDofIndex,
+                                                                              stack.localFlowFlowJacobian[i],
+                                                                              nPDof );
+      RAJA::atomicAdd< parallelDeviceAtomic >( &m_rhs[ dof ], stack.localFlowResidual[i] );
```

Comment:
parallelDeviceAtomic is always employed for this task or this is just an option?